projects/hipblaslt/tensilelite/Tensile/Tests/unit/test_VerifyLRsCompleteBeforeVMFMA.py

```diff
@@ -0,0 +1,340 @@
+    def setUp(self):
+        self.kernel = create_base_kernel()
+        self.num_vmfma = 2 * self.kernel["MIWaveTileA"] * self.kernel["MIWaveTileB"]
+
+    def test_simple_LR0(self):
+        """
+        Verify the simple case where both LRA0 and LRB0 are issued and finished before the halfway point of the main loop.
+        """
+        assert self.num_vmfma == 8
+        
+        optSchedule = {
+            "SYNC": [[3]],
+            "LRA0": [[0, 0]],
+            "LRB0": [[0, 0]]
+        }
+
+        syncCode = [
+            SWaitCnt(dscnt=0, vlcnt=-1, vscnt=-1, comment=""),
+        ]
+        sched = ScheduleInfo(1, self.num_vmfma, optSchedule, syncCode, None, None, None)
+        status, message = verifyLRsCompleteBeforeVMFMA(sched, {"kernel": self.kernel})
+        assert status, f"Schedule should have passed validation but did not. {message}"
+
+        optSchedule["LRA0"] = [[1, 6]]
+        status, message = verifyLRsCompleteBeforeVMFMA(sched, {"kernel": self.kernel})
+        assert not status, f"Schedule should have failed (LRA0 issued after halfway point), but passed."
+        assert message == "Code path 0: LRA0 at index 6 is not valid. Needed before index 5, but only guaranteed at index 3."
+
+        optSchedule["LRA0"] = [[1, 2]]
+        optSchedule["LRB0"] = [[3, 6]]
+        status, message = verifyLRsCompleteBeforeVMFMA(sched, {"kernel": self.kernel})
+        assert not status, f"Schedule should have failed (LRB0 issued after halfway point), but passed."
+        assert message == "Code path 0: LRB0 at index 3 is not valid. Needed before index 4, but only guaranteed at index 3."
+
+    def test_simple_LR0_w_LR1(self):
+        """
+        Handle case where we start reading LRA1 before halfway point.
+        """
+        assert self.num_vmfma == 8
+
+        optSchedule = {
+            "SYNC": [[3, 7]],
+            "LRA0": [[0, 0]],
+            "LRB0": [[0, 0]],
+            "LRA1": [[2]],
+        }
+        syncCode = [
+            SWaitCnt(dscnt=0, vlcnt=-1, vscnt=-1, comment=""),
+            SWaitCnt(dscnt=0, vlcnt=-1, vscnt=-1, comment=""),
+        ]
+        sched = ScheduleInfo(1, self.num_vmfma, optSchedule, syncCode, None, None, None)
+        status, message = verifyLRsCompleteBeforeVMFMA(sched, {"kernel": self.kernel})
+        assert status, f"Schedule should have passed validation 1/2 but did not. {message}"
+
+        # Changing barrier from 0 to 1 for LRA1 should still pass
+        syncCode[0].dscnt = 1
+        status, message = verifyLRsCompleteBeforeVMFMA(sched, {"kernel": self.kernel})
+        assert status, f"Schedule should have passed validation 2/2 but did not. {message}"
+
+    def test_complex_LR0(self):
+        """
+        2nd LRB0 is not needed until iteration 6 & 7, can have SWaitCnt for it after the halfway point.
+        """
+        assert self.num_vmfma == 8
+        
+        optSchedule = {
+            "SYNC": [[3, 4]],
```

Comment:
```suggestion
            "SYNC": [[3, 5]],
```

Based on earlier comment that we go through a faster than b:

index:4 a0 b0
index:5 a1 b0
index:6 a0 b1
index:7 a1 b1

So the sync could be 5? 

Testing at the edge.

Reply:
Yes, I've update the index in the test to reflect that.